libs/prelude/Prelude/Basics.idr

```diff
@@ -25,6 +25,11 @@ id : (1 x : a) -> a           -- Hopefully linearity annotation won't
                               -- break equality proofs involving id
 id x = x
 
+||| Function that duplicates its input
+public export %inline
+dup : a -> (a, a)
```

Comment:
Given this is duplicating its argument, I am not sure we should inline it aggressively.

Reply:
removed %inline